db/migrations/versions/034_geospatial_itl1_region_name.py

```diff
@@ -0,0 +1,68 @@
+"""move GeospatialDim itl1_region_name to its own column
+
+Revision ID: 034_geospatial_itl1_region_name
+Revises: 033_project_geospatial_data
+Create Date: 2024-05-14 10:33:02.419858
+
+"""
+
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy.dialects import postgresql
+
+# revision identifiers, used by Alembic.
+revision = "034_geospatial_itl1_region_name"
+down_revision = "033_project_geospatial_data"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    with op.batch_alter_table("geospatial_dim", schema=None) as batch_op:
+        batch_op.add_column(sa.Column("itl1_region_name", sa.String(), nullable=True))
+
+    op.execute(
+        """
+            UPDATE geospatial_dim
+            SET
+                itl1_region_name = (data_blob ->> 'itl1_region_name')::VARCHAR
+        """
+    )
+
+    op.execute(
+        """
+            UPDATE geospatial_dim
+            SET itl1_region_name = 'Yorkshire and The Humber'
+            WHERE itl1_region_name = 'Yorkshire';
+        """
+    )
+
+    with op.batch_alter_table("geospatial_dim", schema=None) as batch_op:
+        batch_op.alter_column("itl1_region_name", nullable=False)
+        batch_op.drop_column("data_blob")
```

Comment:
Please can we split this out into a separate PR/migration. It could cause errors because the existing app deployment will still be trying to read/write `data_blob` until the code changes in this PR are released.

We could probably get away with deploying this, but that's only because our transaction volume is low. It would be a good habit to get into, to always try to create migrations that would still provide zero-downtime/zero-error guarantees for a higher volume service.

To test locally, essentially run the migration from this PR and then switch back to the main branch. If you get errors with the migration applied and the old code running, it's not backwards compatible.

Reply:
@samuelhwilliams pushed an update to address this, will make a new PR to cover dropping the `data_blob` in `entities.py` and separate migration.